aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+
+Example (class level):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+```
+
+Example (globally):
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+services.AddMvc(options => 
+    options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()));
+```
+
+### IgnoreAntiforgeryToken
+
+The ``IgnoreAntiforgeryToken`` filter is used to eliminate the need for an antiforgery token to be present for a given action (or controller). When applied, this filter will override ``ValidateAntiForgeryToken`` and/or ``AutoValidateAntiforgeryToken`` filters specified at a higher level (globally or on a controller).
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[Authorize]
+[AutoValidateAntiforgeryToken]
+public class ManageController : Controller
+{
+  [HttpPost]
+  [IgnoreAntiforgeryToken]
+  public async Task<IActionResult> DoSomethingSafe(SomeViewModel model)
+  {
+    // no antiforgery token required
+  }
+}
+```
+
+## JavaScript, AJAX, and SPAs
+
+In traditional HTML-based applications, antiforgery tokens are passed to the server using hidden form fields. In modern JavaScript-based apps and single page applications (SPAs), in which AJAX requests are sent programmatically from scripts, other techniques may be used to send the token. If cookies are used to store authentication tokens and to authenticate API requests on the server, then CSRF will be a potential problmem. However, if local storage is used to store the token, CSRF vulnerability may be mitigated, since values from local storage are not sent automatically to the server with every new request.
+
+### AngularJS
+
+AngularJS uses a convention to address CSRF. If the server sends a cookie with the name ``XSRF-TOKEN``, the Angular ``$http`` service will add the value from this cookie to a header when it sends a request to this server. This process is automatic; you don't need to set the header explicitly. The header name is ``X-XSRF-TOKEN``. The server should detect this header and validate its contents. To have your ASP.NET Core API work with this convention, you should configure your app to provide a token in a cookied called ``XSRF-TOKEN``:
```

Comment:
For ASP.NET Core API to work with this convention:

- Configure your app to provide a token in a cookied called ``XSRF-TOKEN``
- onfigure the antiforgery service to look for a header named ``X-XSRF-TOKEN``